### PR TITLE
Add services for listing insights and dashboards

### PR DESCRIPTION
**Pull request by scavnickyj**
---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
// Tiger platform
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

```
// Bear platform
extended test - cypress - integrated
extended test - cypress - isolated
extended test - cypress - record
```

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18069/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19613/ : SUCCESS in 11m 33s

File: libs/sdk-backend-spi/src/workspace/dashboards/index.ts [hunk 1]
```diff
@@ -481,3 +489,59 @@ export interface IWorkspaceDashboardsService {
...
+     */
+    withPage(page: number): IDashboardsQuery;
+
+    /**
+     * Sets filter for the query.
+     *
+     * @param filter - Filtering parameter in RSQL
+     * @returns dashboards query
+     */
+    withFilter(filter: string): IDashboardsQuery;
```
Comment (kandl) [inline libs/sdk-backend-spi/src/workspace/dashboards/index.ts:524]: I don't think leak RSQL outside is a good idea... And what about if we add more filtering options.

File: libs/sdk-backend-spi/src/workspace/insights/index.ts [hunk 2]
```diff
@@ -235,6 +242,55 @@ export interface IGetInsightOptions {
...
+     */
+    withPage(page: number): IInsightsQuery;
+
+    /**
+     * Sets filter for the query.
+     *
+     * @param filter - Filtering parameter in RSQL
+     * @returns insights query
+     */
+    withFilter(filter: string): IInsightsQuery;
```
Comment (kandl) [inline libs/sdk-backend-spi/src/workspace/insights/index.ts:276]: Same thing

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18070/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19614/ : SUCCESS in 11m 24s

**Review by kandl — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4681

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**